types/index.d.ts

```diff
@@ -20,7 +20,7 @@ export type RenderResult<C extends SvelteComponent, Q extends Queries = typeof q
   container: HTMLElement
   component: C
   debug: (el?: HTMLElement | DocumentFragment) => void
-  rerender: (options: SvelteComponentOptions<C>) => void
+  rerender: (options: SvelteComponentOptions<C>) => Promise<void>
```

Comment:
Followup on my other suggestion; I _think_ this would become

```suggestion
  rerender: (options: ComponentProps<C>) => Promise<void>
```